src/libData/AccountData/AccountStore.cpp

```diff
@@ -501,5 +548,289 @@ void AccountStore::RevertCommitTemp() {
+                JSONUtils::GetInstance().convertJsontoStr(map_depth_json));
+          } else if (!JSONUtils::GetInstance().convertStrtoJson(
+                         DataConversion::CharArrayToString(map_depth_data),
+                         map_depth_json)) {
+            LOG_GENERAL(WARNING,
+                        "Account "
+                            << address.hex()
+                            << " failed to parse map_depth_data into json "
+                            << DataConversion::CharArrayToString(
+                                   map_depth_data));
+            return false;
+          }
+          mutable_states.emplace(
+              Contract::ContractStorage2::GetContractStorage()
+                  .GenerateStorageKey(address, FIELDS_MAP_DEPTH_INDICATOR, {}),
+              map_depth_data);
+        } else {
+          LOG_GENERAL(WARNING, "InvokeScillaChecker failed for contract: "
+                                   << address.hex());
+          return false;
+        }
+      } else {
+        LOG_GENERAL(WARNING, "ExportCreateContractFiles failed for contract: "
+                                 << address.hex());
+        return false;
+      }
+    } else {
+      LOG_GENERAL(WARNING,
+                  "Didn't find scilla_version for contract: " << address.hex());
+      return false;
+    }
+
+    /// retrieving mutable states
+    for (const auto& index : account.GetStorageKeyHashes()) {
+      string raw_val = account.GetRawStorage(index, false);
+
+      StateEntry entry;
+      uint32_t version;
+      if (!Messenger::GetStateData(DataConversion::StringToCharArray(raw_val),
+                                   0, entry, version)) {
+        LOG_GENERAL(WARNING, "Messenger::GetStateData failed.");
+        return false;
+      }
+
+      if (version != CONTRACT_STATE_VERSION) {
+        LOG_GENERAL(WARNING, "state data version "
+                                 << version
+                                 << " is not match to CONTRACT_STATE_VERSION "
+                                 << CONTRACT_STATE_VERSION);
+        return false;
+      }
+
+      string tVname = std::get<VNAME>(entry);
+      string tValue = std::get<VALUE>(entry);  /// could be "string" or [map]
+
+      if (!std::get<MUTABLE>(entry)) {
+        continue;
+      }
+
+      string key = i.first.hex();
+      key += SCILLA_INDEX_SEPARATOR + tVname + SCILLA_INDEX_SEPARATOR;
+
+      // Check is the value is map
+      Json::Value json_val;
+
+      if (!map_depth_json.isMember(tVname)) {
+        LOG_GENERAL(WARNING, tVname
+                                 << " is not found in map_depth_json: "
+                                 << JSONUtils::GetInstance().convertJsontoStr(
+                                        map_depth_json));
+        return false;
+      }
+
+      uint map_depth = map_depth_json[tVname].asUInt();
+
+      /// arrayValue & first elemtn contains key/val [TODO]
```

Comment:
I think this comment is no more valid as you're getting `map_depth` from `scilla-checker`